File: chap3/CalculFormel.py
# Ces programmes sont sous licence CeCILL-B V1.
degreMax = 3
t = [0]*(degreMax + 1) 
u = [0]*(degreMax + 1)   
t[3] = 2
t[2] = 8
t[1] = 7
t[0] = 3

# Affichage de la fonction
for degre in range(degreMax + 1):
   print(t[degre],end="")
   if degre != 0:
      print(" x",end="")
      if degre != 1:
         print("^",end="")
         print(degre,end="")
   if degre != degreMax:
      print(" + ",end="")
print()


# Calcul et affichage de sa valeur
x = 5
y = 0
c = 1
for degre in range(degreMax + 1):
   y = y + t[degre] * c;
   c = c * x
print(y)

# Calcul de sa dérivée
for degre in range(degreMax):
   u[degre] = t[degre + 1] * (degre + 1)
u[3] = 0

# Affichage de sa dérivée
for degre in range(degreMax + 1):
   print(u[degre],end="")
   if degre != 0:
      print(" x",end="")
      if degre != 1:
         print("^",end="")
         print(degre,end="")
   if degre != degreMax:
      print(" + ",end="")
print()

File: chap3/CalculFormel.ml
(* Ces programmes sont sous licence CeCILL-B V1. *)

let degreMax = 3

let _ =
  let t = Array.create (degreMax + 1) 0. in
  let u = Array.create (degreMax + 1) 0. in
  t.(3) <- 2.;
  t.(2) <- 8.;
  t.(1) <- 7.;
  t.(0) <- 3.;

(* Affichage de la fonction *)
  for degre = 1 to degreMax do
    print_float t.(degre);
    if degre <> 0 then begin
      print_string " x";
      if degre <> 1 then begin
         print_string "^";
         print_int degre;
      end;
    end;
    if degre <> degreMax then
      print_string " + ";
  done;
  print_newline ();

(* Calcul et affichage de sa valeur *)
  let x = 5. in
  let y = ref 0. in
  let c = ref 1. in
  for degre = 0 to degreMax do
    y := !y +. t.(degre) *. !c;
    c := !c *. x
  done;
  print_float !y; print_newline ();

(* Calcul de sa dérivée *)
for degre = 0 to degreMax-1 do
   u.(degre) <- t.(degre + 1) *. float (degre + 1)
done;
u.(3) <- 0.;

(* Affichage de sa dérivée *)
for degre = 0 to degreMax do
   print_float u.(degre);
  if degre <> 0 then begin
      print_string " x";
      if degre <> 1 then begin
         print_string "^";
         print_int degre;
      end
  end;
  if degre <> degreMax then
      print_string " + ";
done;
print_newline ()
